src/core/render_window.rs

```diff
@@ -177,9 +178,10 @@ where
                     .unwrap()
                     .set_speed_optimization_level(level, &self.speed_optimizer_cache);
                 self.render();
-
-                if level > 0 {
-                    self.render_required = Some(level - 1);
+                if level > 0.0 {
+                    // HACK:  asymtotiallcy approach zero  (maximum quality)
+                    // We'll fix this properly in a follow-up project.
+                    self.render_required = Some(0.5 * level);
                 } else {
                     self.render_required = None;
                 }
```

Comment:
This is the key block here that will be replaced in a follow-up project to use adaptive speed control.